### PR TITLE
feat: change login way and add tracking logic option

### PR DESCRIPTION
**Pull request by leozhao21**
### Changes:

- **Added**:  Logout option
- **Added**:  tracking option
- **Updated**: Use the new login way and remove the old way
- **Updated**: Default show locked problems
- **Fixed**: China points don't open premium problem

### CODE REVIEW

File: src/utils/trackingUtils.ts [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+}
+
+class TrackData implements ITrackData {
+    public reportCache: IReportData[] = [];
+
+    public isSubmit: boolean = false;
+
+    public reportUrl: string = getReportUrl();
+
+    private sendTimer: any;
```
Comment (d0coat01) [inline src/utils/trackingUtils.ts:83]: `private sendTimer: number | undefined`

File: src/utils/trackingUtils.ts [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        _str += _charStr[index];
+    }
+    return _str;
+}
+
+function getAllowReportDataConfig() {
+    const leetCodeConfig: vscode.WorkspaceConfiguration = vscode.workspace.getConfiguration("leetcode");
+    const allowReportData: boolean = !!leetCodeConfig.get<boolean>("allowReportData");
+    return allowReportData;
+}
```
Comment (d0coat01) [inline src/utils/trackingUtils.ts:74]: you don't need to define these typings here, make sure you run `yarn install` and you should see the correct types.

```typescript
    const leetCodeConfig = vscode.workspace.getConfiguration("leetcode");
    const allowReportData = !!leetCodeConfig.get("allowReportData");
    return allowReportData;
```

File: src/utils/trackingUtils.ts [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+    public reportUrl: string = getReportUrl();
+
+    private sendTimer: any;
+
+    constructor() {
+        // setInterval(() => {
+        //   this.submitReport();
+        // }, 10000);
+    }
+
```
Comment (d0coat01) [inline src/utils/trackingUtils.ts:90]: commented out by mistake? or can we remove?

File: src/utils/trackingUtils.ts [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+    public report = (reportItems: IReportData | IReportData[]) => {
+        if (!getAllowReportDataConfig()) return;
+
+        clearTimeout(this.sendTimer);
```
Comment (d0coat01) [inline src/utils/trackingUtils.ts:94]: typeof this.sendTimer === 'number' && clearTimeout(this.sendTimer);

**Review by d0coat01 — APPROVED** [review]
LGTM, some minor typing changes suggested

**Review by Tiffany-yuan — APPROVED** [review]
LGTM

**Pull request merged**